04 - Guides, Workflows, & Courses/Guides/How to add automated tests to your plugin.md

```diff
@@ -1,21 +1,422 @@
+
+```ts
+// example.spec.ts
+import MyPluginLogic from "./myplugin";
+
+describe("MyPlugin Tests", () => {
+  it('should display a static message if the configured limit is exceeded', () => {
+    const mockData = {
+      username: "Bob",
+      numberOfRolls: 10,
+      limit: 8
+    };
+
+    const message = MyPluginLogic.generateDiceRollMessage(mockData);
+
+    expect(message).toEqual("Oh no, you've surpassed your daily limit!");
+  });
+});
+```
+
+This does not test if the notice is really shown, but enables you to test your custom logic that builds the message without the need of potentially complex mocks. You could opt in to manage showing the notice in your extracted function, as well, like shown in the previous example. How you split your code and which parts you want to put under test can be a difficult question to answer and depends on the individual code you're writing, its complexity, and its importance. As mentioned before, combining the presented approaches to test all relevant parts of your plugin is a valid way to go.
+
+## Closing words
+
+While setting up tests might appear tedious at times and is a skill to learn, with a growing code base and growing complexity automated tests can be an enormous help and sometimes the only way to make sure that nothing is seriously broken. It'll give you the confidence to do refactors to your code without the need of extensive manual tests and will also encourage a cleaner structure of productive code for the sake of test-ability.
 
-For now, see the talk [[Plugin Testing for Developers]].
 
 %% Hub footer: Please don't edit anything below this line %%
 
 # This note in GitHub
 
-<span class="git-footer">[Edit In GitHub](https://github.dev/obsidian-community/obsidian-hub/blob/main/04%20-%20Guides%2C%20Workflows%2C%20%26%20Courses/Guides/How%20to%20add%20automated%20tests%20to%20your%20plugin.md "git-hub-edit-note") | [Copy this note](https://raw.githubusercontent.com/obsidian-community/obsidian-hub/main/04%20-%20Guides%2C%20Workflows%2C%20%26%20Courses/Guides/How%20to%20add%20automated%20tests%20to%20your%20plugin.md "git-hub-copy-note") | [Download this vault](https://github.com/obsidian-community/obsidian-hub/archive/refs/heads/main.zip "git-hub-download-vault") </span>
+<span class="git-footer">[Edit In GitHub](https://github.dev/Obsidian-community/Obsidian-hub/blob/main/04%20-%20Guides%2C%20Workflows%2C%20%26%20Courses/Guides/How%20to%20add%20automated%20tests%20to%20your%20plugin.md "git-hub-edit-note") | [Copy this note](https://raw.githubusercontent.com/Obsidian-community/Obsidian-hub/main/04%20-%20Guides%2C%20Workflows%2C%20%26%20Courses/Guides/How%20to%20add%20automated%20tests%20to%20your%20plugin.md "git-hub-copy-note") | [Download this vault](https://github.com/Obsidian-community/Obsidian-hub/archive/refs/heads/main.zip "git-hub-download-vault") </span>
```

Comment:
It's curious that the casing has changed here but the links still work, and will get overwritten on the next weekly automated update, so it's not a problem.

Reply:
That's my fault, sorry for it. I did a find & replace on obsidian to capitalize that - I thought I reverted the changes in the footer but seemingly I did not. Good to know it'll heal itself!